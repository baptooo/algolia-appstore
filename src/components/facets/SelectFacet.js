import React from 'react';
import PropTypes from 'prop-types';
import { FormControl } from 'material-ui/Form';
import Select from 'material-ui/Select';
import { MenuItem } from 'material-ui/Menu';
import Typography from 'material-ui/Typography';

const SelectFacet = ({ name, values, addOrUpdateFacet }) => {
  const currentValue = values.find(({ isRefined }) => isRefined === true);

  return currentValue ? (
    <Typography variant="caption">({currentValue.count} results)</Typography>
  ) : (
    <FormControl style={{ minWidth: "100%" }}>
      <Select
        value={currentValue ? currentValue.name : ''}
        displayEmpty
        onChange={(evt) => addOrUpdateFacet(name, evt.target.value)}
        inputProps={{
          name,
          id: `facet-${name}`,
        }}
      >
        <MenuItem value="">Choose a {name}</MenuItem>
        {values.map(({ name: value }) => (
          <MenuItem key={`facet-${name}-${value}`} value={value}>{value}</MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

SelectFacet.propTypes = {
  name: PropTypes.string.isRequired,
  values: PropTypes.array.isRequired,
  addOrUpdateFacet: PropTypes.func.isRequired
};

export default SelectFacet;