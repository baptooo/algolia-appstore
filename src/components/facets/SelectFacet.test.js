import React from 'react';
import renderer from 'react-test-renderer';
import SelectFacet from "./SelectFacet";

const noop = () => {};
const categories = new Array(5).fill({}).map((facet, idx) => ({
  name: `category-${idx}`,
}));

describe('(Component) SelectFacet', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<SelectFacet name="category" values={categories} addOrUpdateFacet={noop}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});