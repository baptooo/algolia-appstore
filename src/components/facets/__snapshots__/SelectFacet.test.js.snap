// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) SelectFacet should render correctly 1`] = `
<div
  className="MuiFormControl-root-1"
  onBlur={[Function]}
  onFocus={[Function]}
  style={
    Object {
      "minWidth": "100%",
    }
  }
>
  <div
    className="MuiInput-root-10 MuiInput-formControl-11 MuiInput-underline-14"
  >
    <div
      className="MuiSelect-root-5"
    >
      <div
        aria-haspopup="true"
        aria-owns={null}
        aria-pressed="false"
        className="MuiSelect-select-6 MuiSelect-selectMenu-7 MuiInput-input-18"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Choose a 
        category
      </div>
      <input
        aria-invalid={false}
        aria-required={false}
        autoComplete={undefined}
        autoFocus={undefined}
        defaultValue={undefined}
        id="facet-category"
        name="category"
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="hidden"
        value=""
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-23 MuiSelect-icon-9"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
